File: src/main/java/me/lukiiy/discordBridge/discordCmds/Console.java
package me.lukiiy.discordBridge.discordCmds;

import me.lukiiy.discordBridge.DCBridge;
import me.lukiiy.discordBridge.util.GenericHelper;
import me.lukiiy.discordBridge.api.CommandPlate;
import me.lukiiy.discordBridge.util.MemberHelper;
import net.dv8tion.jda.api.entities.Member;
import net.dv8tion.jda.api.interactions.commands.CommandInteraction;
import net.dv8tion.jda.api.interactions.commands.OptionType;
import net.dv8tion.jda.api.interactions.commands.build.CommandData;
import net.dv8tion.jda.api.interactions.commands.build.Commands;
import net.dv8tion.jda.api.interactions.commands.build.OptionData;
import org.bukkit.Bukkit;

public class Console implements CommandPlate {
    @Override
    public CommandData command() {
        return Commands.slash("console", "Executes a command as console.")
                .addOptions(new OptionData(OptionType.STRING, "command", "Command").setRequired(true))
                .setGuildOnly(true);
    }

    @Override
    public void interaction(CommandInteraction i) {
        Member member = i.getMember();
        if (member == null || !MemberHelper.hasRole(member, DCBridge.consoleAdminRole)) {
            i.reply("You don't have access to this command.").setEphemeral(true).queue();
            return;
        }
        String cmd = i.getOption("command").getAsString();
        if (cmd.startsWith("/")) cmd = cmd.substring(1);
        String command = cmd;
        Bukkit.getScheduler().runTask(DCBridge.getInstance(), () -> Bukkit.dispatchCommand(Bukkit.getConsoleSender(), command));
        DCBridge.log(GenericHelper.miniString(DCBridge.configString("format.dcPrefix") + " " + MemberHelper.getCoolName(member) + " executed \"/" + cmd + "\" as console."));
        i.reply("Executed \"" + cmd + "\" as console.").setEphemeral(true).queue();
    }
}

File: minestom/src/main/kotlin/me/lukiiy/discordBridge/discordCmds/Console.kt
package me.lukiiy.discordBridge.discordCmds

import me.lukiiy.discordBridge.Main
import me.lukiiy.discordBridge.api.CommandPlate
import net.dv8tion.jda.api.interactions.InteractionContextType
import net.dv8tion.jda.api.interactions.commands.CommandInteraction
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.CommandData
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.OptionData
import net.minestom.server.MinecraftServer

class Console : CommandPlate {
    override fun command(): CommandData = Commands.slash("console", "Executes a command as console.").addOptions(OptionData(OptionType.STRING, "command", "Command").setRequired(true)).setContexts(InteractionContextType.GUILD, InteractionContextType.BOT_DM)

    override fun interaction(interaction: CommandInteraction) {
        var cmd = interaction.getOption("command")?.asString
        val member = interaction.member

        if (member == null || !Main.context!!.hasConsoleAdminRole(member)) {
            interaction.reply("You don't have access to this command.").setEphemeral(true).queue()
            return
        }

        if (cmd!!.startsWith("/")) cmd = cmd.substring(1)
        val cmdManager = MinecraftServer.getCommandManager()

        cmdManager.dispatcher.execute(cmdManager.consoleSender, cmd)
        MinecraftServer.LOGGER.info(member.effectiveName + " executed \"/" + cmd + "\" as console.")
        interaction.reply("Executed \"$cmd\" as console.").setEphemeral(true).queue()
    }
}
